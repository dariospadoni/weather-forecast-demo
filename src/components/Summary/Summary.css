.summary {
  display: flex;
  color: white;
  padding: 0 48px;
  flex-flow: row wrap;
}

.summary .icon svg { 
  width: 328px; 
  height: 328px;
}

.summary .temperature {
  margin-top: 32px;
  font-weight: 700;
  font-size: 246px;
  line-height: 256px;
  text-align: center;
}

.summary .overline {
  color: #A8AABD;
  font-size: 48px;
  line-height: 56px;
}

.summary .condition {
  padding-right: 140px;
  min-width: 400px;
}

.summary .condition .overline {
  justify-content: space-between;
  display: flex;
}

.summary .icon {
  padding-right: 140px;
}

.summary .date {
  margin-top: 32px;
  font-weight: 700;
  font-size: 100px;
  line-height: 128px;
}

@media (max-width: 1680px) { 
  .summary .condition, .summary .icon {
    padding-right: 100px;
  }
  .summary .condition {
    min-width: 360px;
  }
}

@media (max-width: 1550px) { 
  .summary .city {
    margin-top: 48px;
  }
}

@media (max-width: 1000px) { 
  .summary {
    align-items: center;
    flex-direction: column;
  }
  .summary .condition, .summary .icon {
    padding-right: 0;
  }
  .summary .condition {
    margin-top: 48px;
  }
}

@media (max-width: 640px) { 
  .summary {
    padding: 0 12px; 
  }
}
