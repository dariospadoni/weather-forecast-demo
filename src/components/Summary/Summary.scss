.summary {
  display: flex;
  color: white;
  padding: 0 48px;
  flex-flow: row wrap;
  
  & .summary__icon {
    padding-right: 140px;
   
    svg { 
      width: 328px; 
      height: 328px;
    }
  }
  
  & .summary__overline {
    color: #A8AABD;
    font-size: 48px;
    line-height: 56px;
  }
 
  & .summary__condition {
    padding-right: 140px;
    min-width: 400px;
    
    & .summary__overline--status {
      justify-content: space-between;
      display: flex;
     
    }
    
    & .summary__condition__temperature {
      margin-top: 32px;
      font-weight: 700;
      font-size: 246px;
      line-height: 256px;
      text-align: center;
    }
  }
  
  & .summary__date {
    margin-top: 32px;
    font-weight: 700;
    font-size: 100px;
    line-height: 128px;
  }
} 

@media (max-width: 1680px) { 
  .summary {
    & .summary__icon, & .summary__condition {
      padding-right: 100px;
    }

    &.summary .condition {
      min-width: 360px;
    }
  }
}

@media (max-width: 1550px) { 
  .summary {

    article:last-child {
      margin-top: 48px;
    }
  }
}

@media (max-width: 1024px) { 
  .summary {
    align-items: center;
    flex-direction: column;
  
    & .summary__condition, & .summary__icon {
      padding-right: 0;
    }

    & .summary__condition{
      margin-top: 48px;
    }
  }
}

@media (max-width: 640px) { 
  .summary {
    padding: 12px 0; 
  }
} 
