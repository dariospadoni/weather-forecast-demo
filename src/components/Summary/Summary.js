import React from 'react';
import moment from 'moment';
import PropTypes from 'prop-types';
import './Summary.css';
import getIcon from '../Icons';
import { toCelsius } from '../../utils';


const Summary = ({ weather, cityName }) => {
  const date = moment(weather.date);
  return (
    <main className="summary">
      <div className="icon">
        {getIcon(weather.icon)}
      </div>
      <div className="condition"> 
        <div className="overline">
          <div>
            {weather.status} 
          </div>
          <div>
            {toCelsius(weather.tempMax)} / {toCelsius(weather.tempMin)}
          </div>
        </div>
        <div className="temperature">
          {toCelsius(weather.temp)}
        </div>    
      </div>
      <div className="city"> 
        <div className="overline">
          {cityName}
        </div>
        <div className="date">
          {date.format('dddd')}<br />
          {date.format('D. MMMM')}
        </div>
      </div>
    </main>);
};

Summary.propTypes = {
  cityName: PropTypes.string.isRequired,
  weather: PropTypes.shape({
    icon: PropTypes.string.isRequired,
    status: PropTypes.string.isRequired,
    date: PropTypes.string.isRequired,
    temp: PropTypes.number.isRequired,
    tempMax: PropTypes.number.isRequired,
    tempMin: PropTypes.number.isRequired,
  }).isRequired,
};


export default Summary;
