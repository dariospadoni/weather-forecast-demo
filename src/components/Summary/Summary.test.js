import React from 'react';
import { render } from '@testing-library/react';
import Summary from './Summary';

const mockData = {
  cityName: 'Berlin',
  weather: {
    icon: '01d',
    status: 'Clean',
    date: '2020-08-04 18:00:00',
    temp: 293.55,
    tempMax: 294.05,
    tempMin:  293.55,
  }
}

it('renders correctly', () => {
  const component = render(<Summary cityName={mockData.cityName} weather={mockData.weather} />);
  expect(component).toMatchSnapshot();
});
