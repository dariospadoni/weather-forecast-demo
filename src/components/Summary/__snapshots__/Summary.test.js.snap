// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="summary"
      >
        <article
          class="summary__icon"
        >
          <svg>
            weather-sun.svg
          </svg>
        </article>
        <article
          class="summary__condition"
        >
          <div
            class="summary__overline summary__overline--status"
          >
            <div>
              Clean
            </div>
            <div>
              21°
               / 
              20°
            </div>
          </div>
          <div
            class="summary__condition__temperature"
          >
            20°
          </div>
        </article>
        <article>
          <div
            class="summary__overline"
          >
            Berlin
          </div>
          <div
            class="summary__date"
          >
            Tuesday
            <br />
            4. August
          </div>
        </article>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="summary"
    >
      <article
        class="summary__icon"
      >
        <svg>
          weather-sun.svg
        </svg>
      </article>
      <article
        class="summary__condition"
      >
        <div
          class="summary__overline summary__overline--status"
        >
          <div>
            Clean
          </div>
          <div>
            21°
             / 
            20°
          </div>
        </div>
        <div
          class="summary__condition__temperature"
        >
          20°
        </div>
      </article>
      <article>
        <div
          class="summary__overline"
        >
          Berlin
        </div>
        <div
          class="summary__date"
        >
          Tuesday
          <br />
          4. August
        </div>
      </article>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
